src/transformers/cache_utils.py

```diff
@@ -1182,10 +1187,24 @@ def update(
 
         cache_position = cache_kwargs.get("cache_position")
 
-        k_out = self.key_cache[layer_idx]
-        v_out = self.value_cache[layer_idx]
-        key_states = key_states.to(k_out.dtype)
-        value_states = value_states.to(v_out.dtype)
+        if self.key_cache[layer_idx].device.type == "meta":
+            k_out = torch.zeros(
+                *self.key_cache[layer_idx].size(),
+                device=key_states.device,
+                dtype=key_states.dtype,
+            )
```

Comment:
```suggestion
            k_out = torch.zeros_like(self.key_cache[layer_idx], device=key_states.device)
```
let's use torch primitives !

Reply:
also why don't we just always do:
```python 
            self.key_cache[layer_idx].to(key_states.device, key_states.dtype)
            self.value_cache[layer_idx].to(key_states.device, key_states.dtype)
```
this would put it from "meta" to "device" or will be a no op if this is already the case. Seems like we are overcomplicating here

Reply:
for `meta` we can't just move with `to(device)`, but the first option is defi simpler. For some reason I thought `zeros_like` would neither work with `meta` hehe

Reply:
Right because of `NotImplementedError: Cannot copy out of meta tensor; no data!` but yeah let's be simpler in general!